packages/story-editor/src/elements/video/trim.js

```diff
@@ -89,17 +117,35 @@ function VideoTrim({ box, element }) {
     [setVideoNode]
   );
 
+  // If there's an original video, we should use that for trimming instead.
+  if (!isFetchingResource && shouldFetchResource({ end, start, original })) {
+    const endTime = getMsFromHMS(end);
+    const startTime = getMsFromHMS(start);
+    fetchOriginalResource({ original, endTime, startTime });
+    return (
+      <Wrapper>
+        <Spinner>
+          <CircularProgress />
+        </Spinner>
+      </Wrapper>
+    );
+  }
```

Comment:
Calling a function with side-effects (`fetchOriginalResource` in particular) directly in the render function is bad karma (can have unforeseen consequences) and even throws a warning:

```
Warning: Cannot update a component (`VideoTrimProvider`) while rendering a different component (`VideoTrim`).
```

The loader return should probably just be based on `isFetchingResource` being true (or maybe a combination of `isFetchingResource || shouldFetchResource`).

The call to `fetchOriginalResource` should be moved inside a (properly guarded) `useEffect`.

But I'd still like to see this call moved to the provider. The provider knows when the trim mode is triggered and can already then determine if a video is to be loaded. This component actually doesn't need to know anything about original videos or not - it'll just display whatever video that the provider tells it to. The provider should abstract out where it's the "real" video or an "original" video. The same goes for the video trimmer bar.